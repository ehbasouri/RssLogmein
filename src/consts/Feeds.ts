function rndYear(): number {
  return 2000 + Math.floor(Math.random() * 20) + 1;
}

export const FeedMock = {
  title: 'Feed Title',
  description: 'This is my personal feed!',
  id: 'http://example.com/',
  link: 'http://example.com/',
  language: 'en', // optional, used only in RSS 2.0, possible values: http://www.w3.org/TR/REC-html40/struct/dirlang.html#langcodes
  image: 'http://example.com/image.png',
  favicon: 'http://example.com/favicon.ico',
  copyright: 'All rights reserved 2013, John Doe',
  updated: new Date(rndYear(), 6, 14), // optional, default = today
  generator: 'awesome', // optional, default = 'Feed for Node.js'
  feedLinks: {
    json: 'https://example.com/json',
    atom: 'https://example.com/atom',
  },
  author: {
    name: 'John Doe',
    email: 'johndoe@example.com',
    link: 'https://example.com/johndoe',
  },
};

export const FeedPosts = [
  {
    url: 'https://example.com/posts1',
    title: 'post1',
    description: 'description1',
    content: '',
    date: new Date(),
    image: '',
  },
  {
    url: 'https://example.com/posts2',
    title: 'post1',
    description: 'description2',
    content: '',
    date: new Date(),
    image: '',
  },
];

export const FeedItem = {
  author: [
    {
      name: 'Jane Doe',
      email: 'janedoe@example.com',
      link: 'https://example.com/janedoe',
    },
    {
      name: 'Joe Smith',
      email: 'joesmith@example.com',
      link: 'https://example.com/joesmith',
    },
  ],
  contributor: [
    {
      name: 'Shawn Kemp',
      email: 'shawnkemp@example.com',
      link: 'https://example.com/shawnkemp',
    },
    {
      name: 'Reggie Miller',
      email: 'reggiemiller@example.com',
      link: 'https://example.com/reggiemiller',
    },
  ]
}
