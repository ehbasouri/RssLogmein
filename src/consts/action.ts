export const UPDATE_GENERAL_PROPS: string = 'UPDATE_GENERAL_PROPS';
export const RESET_GENERAL_PROPS: string = 'RESET_GENERAL_PROPS';
