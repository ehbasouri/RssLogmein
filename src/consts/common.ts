export const FAVOURITE_LIST: string = 'FAVOURITE_LIST';
