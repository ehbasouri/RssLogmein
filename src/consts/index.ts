export * from './screen';
export * from './action';
export * from './Feeds';
