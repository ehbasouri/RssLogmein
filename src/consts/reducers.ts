// export const FAVOURITE_LIST = 'favouriteFeeds';