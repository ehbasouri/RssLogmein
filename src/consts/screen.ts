export const FEED_LIST: string = 'feed_list';
export const ENTRY: string = 'entry';
export const FAVOURITE_LIST = 'favourite_list';
