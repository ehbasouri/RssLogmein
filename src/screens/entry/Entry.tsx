import React, {useContext} from 'react';
import {FlatList, View} from 'react-native';
import {AppContext} from '../../App/AppContext';
import {EntryNavigationProps} from '../../App/AppRouter';
import {RssHeader} from '../../components/RssHeader';
import {EntryItem} from './item';

export default function Entry({navigation, route}: EntryNavigationProps) {
  const {onEntryPress} = useContext(AppContext);
  const {items, title} = route.params;
  function onBackPress() {
    navigation.goBack();
  }

  return (
    <View>
      <FlatList
        ListHeaderComponent={
          <RssHeader showBack onBackPress={onBackPress} title={title} />
        }
        data={items}
        renderItem={({item}) => {
          return <EntryItem onPress={onEntryPress} entry={item} />;
        }}
        keyExtractor={(item, index) => index.toString()}
        stickyHeaderIndices={[0]}
      />
    </View>
  );
}
