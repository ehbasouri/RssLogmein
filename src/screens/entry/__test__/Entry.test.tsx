import {render} from '@testing-library/react-native';
import React from 'react';
import {FeedItem, FeedPosts} from '../../../consts/Feeds';
import Entry from '../Entry';

const newEntry = {
  title: FeedPosts[0].title,
  id: FeedPosts[0].url,
  link: FeedPosts[0].url,
  description: FeedPosts[0].description,
  content: FeedPosts[0].content,
  date: FeedPosts[0].date,
  image: FeedPosts[0].image,
  ...FeedItem,
};

it('Entry tests', () => {
  let props: any = {
    route: {
      params: {
        options: newEntry,
      },
    },
    navigation: {
      setOptions: jest.fn(),
    },
  };
  const {getByText} = render(<Entry {...props} />);

  const header = getByText(newEntry.title);
  expect(header).toBeTruthy();
});
