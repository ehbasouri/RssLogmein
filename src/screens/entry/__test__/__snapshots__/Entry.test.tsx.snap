// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test suite fetch and display data in FeedList 1`] = `
<View>
  <RCTScrollView
    ListHeaderComponent={
      <RssHeader
        onBackPress={[Function]}
        showBack={true}
        title="title"
      />
    }
    data={
      Array [
        Object {
          "links": Array [
            Object {
              "url": "url",
            },
          ],
          "title": "item-title",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingBottom": 16,
              "paddingHorizontal": 6,
              "paddingTop": 50,
            }
          }
        >
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "blue",
                "flex": 1,
              }
            }
          >
            BACK
          </Text>
          <Text
            style={
              Object {
                "flex": 4,
                "fontSize": 18,
                "fontWeight": "bold",
                "width": 40,
              }
            }
          >
            title
          </Text>
          <Text
            style={
              Object {
                "color": "blue",
                "flex": 1,
              }
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#C1C1C1",
              "borderWidth": 0.5,
              "marginTop": 20,
              "padding": 20,
            }
          }
        >
          <Text>
            item-title
          </Text>
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "blue",
              }
            }
            testID="entry-item"
          >
            url
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
